### PR TITLE
Add Kubecon page for KubeCon Paris 2024

### PR DESCRIPTION
**Pull request by kingdonb**
The deploy preview is ready at:

https://deploy-preview-1804--fluxcd.netlify.app/kubecon/

How this is supposed to work, you can just edit the Google Sites page, and this workflow in:

* fluxcd/community#349

...comes along every 6 hours, wget's the html and feeds it through `html2md`, a Go tool that does a good job of converting HTML into Markdown

Then it runs a bunch of string substitutions on the Markdown output, to make sure we can serve the images from fluxcd.io instead of hot-linking Google Sites. The images all have some semantic associations that are conveyed through alt tags. We can change anything here, but I think it's pretty good for a first pass, and it should only open a Pull Request when there is a substantive change on the Sites.

Then, further string substitutions ensure that the things which should be shown as columns are given semantic containers so we can use CSS to turn them into columns.

We can trigger the workflow every 6 hours (we can shorten this interval during Kubecon - the workflow is fast) or you can trigger it manually, through the workflow dispatch.

The script has a debug mode you can enable with `export DEBUG=1` - the instructions to operate the script are in the header, if you run the script on MacOS, then it will try to use `gsed` and `ghead` that are like the sed and head commands we'll find on our GitHub Actions runner workflow.

### CODE REVIEW

**Comment by kingdonb** [comment]
I marked this one as ready for review, but we should not merge it before:

* https://github.com/fluxcd/community/pull/349

(and the preview deploy will fail unless that one is merged first.)

**Comment by kingdonb** [comment]
There's a workflow included in fluxcd/community#349 now that does the automatic pull from @mewzherder Google Site and populates a PR. Because this workflow is automated, @mewzherder doesn't need a second reviewer to merge a PR to the site - I think this addresses the main issue for Kubecon. If you can give me access to the Google Sites page then I'll make the necessary updates to the images in this PR, so they are hard links to local static content.

(Look at the generated PR to see what I mean: there's some dynamic cache key thing that changes depending on which Google site cache serves up the page. That will make the PRs spammy and annoying, so let's fix it... I know what it needs)

**Comment by kingdonb** [comment]
I made the necessary changes in the Google Sites page upstream to support discerning the images and differentiating them with CSS classes, using the `figure` shortcode.

We should be able to add some CSS to the page once everything is connected and preview how it's going to look here. 🚀

**Comment by kingdonb** [comment]
I added a description to this PR for reviewers 👍 

Still TODO before this one can merge:
- [ ] CSS for the classes named in alt tags
- [ ] flux-logo-inner-header-left Flux
- [ ] inner-header-right-align Blob waving
- [ ] stickers-float-left

Just confirming that these are all original works and we have the copyright permission from the original creator of this site @mewzherder to scrape it and reblog it on fluxcd.io/kubecon ✅ 

Then LGTM after fluxcd/community#349 is merged

**Comment by kingdonb** [comment]
Wow, it was not easy, but:

https://deploy-preview-1804--fluxcd.netlify.app/kubecon/

replicates the structure of

https://sites.google.com/view/flux-kubecon-paris-2024/home

🎉 

Edits to the Google Sites page will result in some automation converting it from HTML into Markdown, and then partially back again, finally culminating in a pull-request auto-filing itself against this file:

https://github.com/fluxcd/community/blob/main/KUBECON.md

Any maintainer to the Community repo can approve and merge it, no second approver is needed

This PR pulls that file in via the existing `external-sources` magic, then the Website repo, on its next refresh, will intake the KUBECON.md and place it at `content/en/kubecon.md` where it as served as https://fluxcd.io/kubecon

🥅 ⚽ 👣 

I think this PR is ready to consider for merge. We can make improvements to the CSS if anyone has feedback on that, but replicating the column format after html2md was quite a trick. I hope we can use this.

**Review by scottrigby — APPROVED** [review]
looks super good!

**Comment by kingdonb** [comment]
The other benefit here is that being in an external source, changes to KUBECON.md do not need to undergo a second PR to backport them to the website. We could trigger the every-6-hours job across repositories, to quicken the response time from the two jobs that have to run in order, (since the goal was to be able to push a change out in less than 10 minutes I think we should do that - it's not much extra work)

I'll look into what else we would need to do. But if there are no objections, I think we can make this page live on Friday?

**Comment by kingdonb** [comment]
There is one error I just spotted:

```
[the podinfo sample app](https://github.com/stefanprodan/podinfo)
```

This markdown link occurs inside an HTML block, so it doesn't get processed through markdown.

Fixed by fluxcd/community#362

**Review by kingdonb — COMMENTED** [review]
TODO:

Follow-up notes in

- [ ] fluxcd/community#362

(we should be able to trigger an immediate refresh by filtering the push: branches: main event for changes to any of the three files listed in external-sources)

but that's a small quality-of-life enhancement which will not take much time to implement. We need the site to go live today as planned, so we can begin testing the QR code, testing the workflow in production...

And there are no errors left in the copy as far as I can tell, 👍 LGTM

**Pull request merged**
